run_e3sm.template.sh

```diff
@@ -245,18 +247,33 @@ create_newcase() {
 
     echo $'\n----- Starting create_newcase -----\n'
 
-    ${CODE_ROOT}/cime/scripts/create_newcase \
-        --case ${CASE_NAME} \
-        --case-group ${CASE_GROUP} \
-        --output-root ${CASE_ROOT} \
-        --script-root ${CASE_SCRIPTS_DIR} \
-        --handle-preexisting-dirs u \
-        --compset ${COMPSET} \
-        --res ${RESOLUTION} \
-        --machine ${MACHINE} \
-        --project ${PROJECT} \
-        --walltime ${WALLTIME} \
-        --pecount ${PELAYOUT}
+	if [[ -z "$CASE_GROUP" ]]; then
+		${CODE_ROOT}/cime/scripts/create_newcase \
+			--case ${CASE_NAME} \
+			--output-root ${CASE_ROOT} \
+			--script-root ${CASE_SCRIPTS_DIR} \
+			--handle-preexisting-dirs u \
+			--compset ${COMPSET} \
+			--res ${RESOLUTION} \
+			--machine ${MACHINE} \
+			--project ${PROJECT} \
+			--walltime ${WALLTIME} \
+			--pecount ${PELAYOUT}
+	else
+		${CODE_ROOT}/cime/scripts/create_newcase \
+			--case ${CASE_NAME} \
+			--case-group $(CASE_GROUP) \
```

Comment:
Why the change from `{}` to `()`?

Reply:
That was another way for handling empty string. We don't need it now as we are doing an explicit check. I will remove.